docs/tests.rst

```diff
@@ -167,33 +167,91 @@ Here are a few tips for writing tests:
 .. _Mocha: https://mochajs.org/
 
 
-Smoketests
-==========
+Functional UI Tests
+===================
 
-We can do more comprehensive front-end testing with the smoketests.
-They're located in the ``smoketests/`` directory. See the ``README.rst``
-file in that directory for how to set them up as well as other details.
+We can do more comprehensive front-end testing with the functional UI tests.
+They're located in the ``tests/functional`` directory.
 
+Installing dependencies
+-----------------------
 
-.. _tests-chapter-qa-test-suite:
+Followed the steps in :ref:`the installation docs <hacking-howto-chapter>`,
+including the test dependencies to make sure you have everything you need to
+run the tests. If you're running the tests against a deployed environment then
+there's no need to install anything other than the test dependencies.
 
-The QA test suite
-=================
+Create test users
+-----------------
+
+Some of the tests require logging in as a administrator, and others require
+logging in as a user. To run these tests you will need to create accounts in
+the target environment. If you're running against a local instance of the
+application you can create these users by running the following script::
+
+  $ ./manage.py shell < ./scripts/create_user_and_superuser.py
```

Comment:
When attempting to change this I hit the following exception, which I can only guess is safely ignored in the current version:

```
$ ./manage.py runscript create_user_and_superuser
Exception while running run() in 'scripts.create_user_and_superuser'
Traceback (most recent call last):
  File "./manage.py", line 32, in <module>
    execute_from_command_line(sys.argv)
  File "/home/dhunt/kitsune/venv/local/lib/python2.7/site-packages/django/core/management/__init__.py", line 385, in execute_from_command_line
    utility.execute()
  File "/home/dhunt/kitsune/venv/local/lib/python2.7/site-packages/django/core/management/__init__.py", line 377, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/home/dhunt/kitsune/venv/local/lib/python2.7/site-packages/django/core/management/base.py", line 288, in run_from_argv
    self.execute(*args, **options.__dict__)
  File "/home/dhunt/kitsune/venv/local/lib/python2.7/site-packages/django/core/management/base.py", line 338, in execute
    output = self.handle(*args, **options)
  File "/home/dhunt/kitsune/venv/local/lib/python2.7/site-packages/django_extensions/management/commands/runscript.py", line 147, in handle
    run_script(mod, *script_args)
  File "/home/dhunt/kitsune/venv/local/lib/python2.7/site-packages/django_extensions/management/commands/runscript.py", line 74, in run_script
    mod.run(*script_args)
  File "/home/dhunt/kitsune/scripts/create_user_and_superuser.py", line 33, in run
    enter_data(data)
  File "/home/dhunt/kitsune/scripts/create_user_and_superuser.py", line 26, in enter_data
    create_user(username=user['username'], password=user['password'], email=user['email'])
  File "/home/dhunt/kitsune/scripts/create_user_and_superuser.py", line 18, in create_user
    User.objects.create_user(username=username, password=password, email=email)
  File "/home/dhunt/kitsune/venv/local/lib/python2.7/site-packages/django/contrib/auth/models.py", line 183, in create_user
    **extra_fields)
  File "/home/dhunt/kitsune/venv/local/lib/python2.7/site-packages/django/contrib/auth/models.py", line 178, in _create_user
    user.save(using=self._db)
  File "/home/dhunt/kitsune/venv/local/lib/python2.7/site-packages/django/db/models/base.py", line 589, in save
    force_update=force_update, update_fields=update_fields)
  File "/home/dhunt/kitsune/venv/local/lib/python2.7/site-packages/django/db/models/base.py", line 626, in save_base
    update_fields=update_fields, raw=raw, using=using)
  File "/home/dhunt/kitsune/venv/local/lib/python2.7/site-packages/django/dispatch/dispatcher.py", line 198, in send
    response = receiver(signal=self, sender=sender, **named)
  File "/home/dhunt/kitsune/kitsune/search/models.py", line 232, in update
    maybe_call_method(instance, kw.get('raw'), 'index_later')
  File "/home/dhunt/kitsune/kitsune/search/models.py", line 226, in maybe_call_method
    obj = instance_to_indexee(instance)
  File "/home/dhunt/kitsune/kitsune/users/models.py", line 312, in get_profile
    return Profile.objects.get(user=u)
  File "/home/dhunt/kitsune/venv/local/lib/python2.7/site-packages/django/db/models/manager.py", line 92, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/home/dhunt/kitsune/venv/local/lib/python2.7/site-packages/django/db/models/query.py", line 351, in get
    num = len(clone)
  File "/home/dhunt/kitsune/venv/local/lib/python2.7/site-packages/django/db/models/query.py", line 122, in __len__
    self._fetch_all()
  File "/home/dhunt/kitsune/venv/local/lib/python2.7/site-packages/django/db/models/query.py", line 966, in _fetch_all
    self._result_cache = list(self.iterator())
  File "/home/dhunt/kitsune/venv/local/lib/python2.7/site-packages/django/db/models/query.py", line 265, in iterator
    for row in compiler.results_iter():
  File "/home/dhunt/kitsune/venv/local/lib/python2.7/site-packages/django/db/models/sql/compiler.py", line 701, in results_iter
    for rows in self.execute_sql(MULTI):
  File "/home/dhunt/kitsune/venv/local/lib/python2.7/site-packages/django/db/models/sql/compiler.py", line 787, in execute_sql
    cursor.execute(sql, params)
  File "/home/dhunt/kitsune/venv/local/lib/python2.7/site-packages/django/db/backends/utils.py", line 81, in execute
    return super(CursorDebugWrapper, self).execute(sql, params)
  File "/home/dhunt/kitsune/venv/local/lib/python2.7/site-packages/django/db/backends/utils.py", line 65, in execute
    return self.cursor.execute(sql, params)
  File "/home/dhunt/kitsune/venv/local/lib/python2.7/site-packages/django/db/utils.py", line 94, in __exit__
    six.reraise(dj_exc_type, dj_exc_value, traceback)
  File "/home/dhunt/kitsune/venv/local/lib/python2.7/site-packages/django/db/backends/utils.py", line 65, in execute
    return self.cursor.execute(sql, params)
  File "/home/dhunt/kitsune/venv/local/lib/python2.7/site-packages/django/db/backends/mysql/base.py", line 129, in execute
    return self.cursor.execute(query, args)
  File "/home/dhunt/kitsune/venv/local/lib/python2.7/site-packages/MySQLdb/cursors.py", line 173, in execute
    self.errorhandler(self, exc, value)
  File "/home/dhunt/kitsune/venv/local/lib/python2.7/site-packages/MySQLdb/connections.py", line 36, in defaulterrorhandler
    raise errorclass, errorvalue
django.db.utils.OperationalError: (1054, "Unknown column 'users_profile.csat_email_sent' in 'field list'")
```